### PR TITLE
DRE EntityCSMoveAlucard

### PR DESCRIPTION
**Pull request by sonicdcer**
Had to declare D_80097400 before each function use because it needs to be a different type, may be part of a struct.

### CODE REVIEW

File: src/st/dre/14774.c [hunk 1]
```diff
@@ -137,7 +137,140 @@ void func_801961DC(s16 arg0) {
...
+        }
+        self->ext.generic.unk7C.s = 128;
+        break;
+
+    case 1:
+        g_Player.D_80072EF4 = 0;
+        if (self->ext.generic.unk7C.u != 0) {
+            self->ext.generic.unk7C.u--;
+        }
+        if (g_Player.unk0C & 7) {
```
Comment (sozud) [inline src/st/dre/14774.c:171]: I tried a couple of simplifications here but they didn't match. This seems to be the scratch https://decomp.me/scratch/T8Auk

**Review by sozud — APPROVED** [review]
Looks good

File: src/st/dre/14774.c [hunk 2]
```diff
@@ -137,7 +137,140 @@ void func_801961DC(s16 arg0) {
...
+    }
+
+    switch (self->step) {
+    case 0:
+        InitializeEntity(D_8018047C);
+        D_8003C8B8 = 0;
+        D_80097400 = 1;
+        g_Player.D_80072EF4 = 0;
+        g_Player.D_80072EFC = 1;
+        if (g_DemoMode != Demo_None) {
```
Comment (Xeeynamo) [inline src/st/dre/14774.c:159]:
```suggestion
        if (g_DemoMode != Demo_None) {
            self->ext.generic.unk7C.s = 64;
        } else {
            self->ext.generic.unk7C.s = 128;
        }
        break;
```

File: src/st/dre/14774.c [hunk 3]
```diff
@@ -137,7 +137,140 @@ void func_801961DC(s16 arg0) {
...
+        break;
+
+    case 1:
+        g_Player.D_80072EF4 = 0;
+        if (self->ext.generic.unk7C.u != 0) {
+            self->ext.generic.unk7C.u--;
+        }
+        if (g_Player.unk0C & 7) {
+            if (g_blinkTimer & 1) {
+                if (!(g_Player.unk0C & 1)) {
```
Comment (Xeeynamo) [inline src/st/dre/14774.c:173]:
```suggestion
            if (g_blinkTimer & 1) {
                if (g_Player.unk0C & 1) {
                    g_Player.D_80072EF4 = 8;
                } else if (g_Player.unk0C & 2) {
                    g_Player.D_80072EF4 = 4;
                } else if (g_Player.unk0C & 4) {
                    g_Player.D_80072EF4 = 2;
                }
            }
```

File: src/st/dre/14774.c [hunk 4]
```diff
@@ -137,7 +137,140 @@ void func_801961DC(s16 arg0) {
...
+    case 5: // Conversation with fake lisa
+        g_Player.D_80072EFC = 1;
+        if (D_801A3F84 & 8) {
+            SetStep(6);
+        }
+        break;
+
+    case 6: // Alucard's reaction to realizing she's not his mother
+        g_Player.D_80072EFC = 1;
+        if (AnimateEntity(D_80180944, self) == 0) {
```
Comment (Xeeynamo) [inline src/st/dre/14774.c:244]:
```suggestion
        if (AnimateEntity(D_80180944, self) == 0) {
            SetStep(7);
        } else {
            player->animCurFrame = self->animCurFrame;
        }
        break;
```

**Review by Xeeynamo — CHANGES_REQUESTED** [review]

**Comment by sozud** [comment]
@Xeeynamo This one is waiting on figuring out a better solution for `D_80097400`, correct? I think you mentioned you wanted to take a look, did you have a chance yet?

**Comment by Xeeynamo** [comment]
Correct, we need to find how that symbol is used and use it accordingly. I have not yet had the opportunity to dive into it yet.

**Pull request closed without merging**